service.py

```diff
@@ -51,12 +51,12 @@ def _start(self):
         util.LOG('DOWNLOAD SERVICE: START')
         info = self.getNextQueuedDownload()
 
-        while info and not xbmc.abortRequested:
+        while info and not xbmc.Monitor().abortRequested():
```

Comment:
since this is called in a loop it's usually a good practice to define the monitor above and reuse the instance below. Otherwise you're creating a Monitor() instance all the time

Reply:
I will modify it. Thank you!